src/datasets/domain/useCases/validators/NewDatasetValidator.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  NewDataset,
+  NewDatasetMetadataFieldValue,
+  NewDatasetMetadataChildFieldValue,
+  NewDatasetMetadataBlockValues,
+} from '../../models/NewDataset';
+import { NewResourceValidator } from '../../../../core/domain/useCases/validators/NewResourceValidator';
+import { MetadataFieldInfo, MetadataBlock } from '../../../../metadataBlocks';
+import { ResourceValidationError } from '../../../../core/domain/useCases/validators/errors/ResourceValidationError';
+import { EmptyFieldError } from './errors/EmptyFieldError';
+import { FieldValidationError } from './errors/FieldValidationError';
+import { ControlledVocabularyFieldError } from './errors/ControlledVocabularyFieldError';
+import { DateFormatFieldError } from './errors/DateFormatFieldError';
+
+export interface NewDatasetMetadataFieldAndValueInfo {
+  metadataFieldInfo: MetadataFieldInfo;
+  metadataFieldKey: string;
+  metadataFieldValue: NewDatasetMetadataFieldValue;
+  metadataBlockName: string;
+  metadataParentFieldKey?: string;
+  metadataFieldPosition?: number;
+}
+
+export class NewDatasetValidator implements NewResourceValidator {
+  async validate(resource: NewDataset, metadataBlocks: MetadataBlock[]): Promise<void | ResourceValidationError> {
```

Comment:
I don't understand why this is an async method if it doesn't make any asynchronous calls and doesn't return anything. It seems to me as a completely synchronous method

We could be calling the validate method synchronously in the use case

```
async execute(newDataset: NewDataset, collectionId: string = 'root'): Promise<CreatedDatasetIdentifiers> {
    const metadataBlocks = await this.getNewDatasetMetadataBlocks(newDataset);

    this.newDatasetValidator.validate(newDataset, metadataBlocks);
    
    return this.datasetsRepository.createDataset(newDataset, metadataBlocks, collectionId);
}
```